base_station/gui/src/components/PDBFuse.vue

```diff
@@ -0,0 +1,305 @@
+        </tr>
+      </tbody>
+    </table>
+  </div>
+</div>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      pdb_data: {
+        three: {
+          temp: {val: 0.0, color: "grey"},
+          current: {val: 0.0, color: "grey"},
+          voltage: {val: 0.0, color: "grey"}
+        },
+        five: {
+          temp: {val: 0.0, color: "grey"},
+          current: {val: 0.0, color: "grey"},
+          voltage: {val: 0.0, color: "grey"}
+        },
+        twelve: {
+          temp: {val: 0.0, color: "grey"},
+          current: {val: 0.0, color: "grey"},
+          voltage: {val: 0.0, color: "grey"}
+        }
+      },
+
+      fuse_data: {
+        cytron1: {
+          current: {val: 0.0, color: "grey"},
+          voltage: {val: 0.0, color: "grey"}
+        },
+        cytron2: {
+          current: {val: 0.0, color: "grey"},
+          voltage: {val: 0.0, color: "grey"}
+        },
+        cytron3: {
+          current: {val: 0.0, color: "grey"},
+          voltage: {val: 0.0, color: "grey"}
+        },
+        cytron4: {
+          current: {val: 0.0, color: "grey"},
+          voltage: {val: 0.0, color: "grey"}
+        },
+        cytron5: {
+          current: {val: 0.0, color: "grey"},
+          voltage: {val: 0.0, color: "grey"}
+        },
+        cytron6: {
+          current: {val: 0.0, color: "grey"},
+          voltage: {val: 0.0, color: "grey"}
+        },
+        nichrome1: {
+          current: {val: 0.0, color: "grey"},
+          voltage: {val: 0.0, color: "grey"}
+        },
+        nichrome2: {
+          current: {val: 0.0, color: "grey"},
+          voltage: {val: 0.0, color: "grey"}
+        },
+        nichrome3: {
+          current: {val: 0.0, color: "grey"},
+          voltage: {val: 0.0, color: "grey"}
+        },
+        hbridge1: {
+          current: {val: 0.0, color: "grey"},
+          voltage: {val: 0.0, color: "grey"}
+        },
+        hbridge2: {
+          current: {val: 0.0, color: "grey"},
+          voltage: {val: 0.0, color: "grey"}
+        },
+      }
+    }
+  },
+
+  created: function() {
+
+    this.$parent.subscribe('/pdb_data', (msg) => {
+      this.updateVal(this.pdb_data.three.temp, msg.temp[0], 100)
+      this.updateVal(this.pdb_data.three.current, msg.current[0], 2)
+      this.updateVal(this.pdb_data.three.voltage, msg.voltage[0], 3.6)
+      
+      this.updateVal(this.pdb_data.five.temp, msg.temp[1], 100)
+      this.updateVal(this.pdb_data.five.current, msg.current[1], 8)
+      this.updateVal(this.pdb_data.five.voltage, msg.voltage[1], 5.3)
+      
+      this.updateVal(this.pdb_data.twelve.temp, msg.temp[2], 100)
+      this.updateVal(this.pdb_data.twelve.current, msg.current[2], 8)
+      this.updateVal(this.pdb_data.twelve.voltage, msg.voltage[2], 14)
+    })
+
+    this.$parent.subscribe('/fuse_data', (msg) => {
+      this.updateVal(this.fuse_data.cytron1.current, msg.current[0], 2.5)
```

Comment:
anyway to make this section less copy-paste-y?

Reply:
Somewhat better now. The data section for this file is still kinda bad but I couldn't figure out how to have multiple instances of data without code duplication there.